src/entity.ts

```diff
@@ -991,16 +991,6 @@ export namespace entity {
     const MAX_DATASTORE_VALUE_LENGTH = 1500;
     if (Array.isArray(entities)) {
       for (const entry of entities) {
-        if (entry && entry.name && entry.value) {
```

Comment:
The original intent of this `if` statement was to deal with arrays passed into the save function. When arrays are passed into the save function, they are normally expected to have `name` and `value` properties defined at the top level and then [their proto is encoded using the name as the key](https://github.com/googleapis/nodejs-datastore/blob/519e33b3c69c64cfcd4a349c507bf9de85697fdb/src/index.ts#L1143-L1164) so this code adds `excludeFromIndexes:true` in the right places for this special encoding.

The problem with this if statement is that this function is recursive so this logic will be applied recursively, but we only want this logic to be applied at the top level for arrays because the special encoding is only done for arrays at the top level.